src/config/mongoose_config_parser_toml.erl

```diff
@@ -471,30 +472,700 @@ process_module([Mod|_] = Path, Opts) ->
+    [{ldap_gfilter, b2l(V)}];
+module_opt([<<"ldap_ufilter">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_ufilter, b2l(V)}];
+module_opt([<<"buffer_max">>, <<"mod_stream_management">>|_], V) ->
+    [{buffer_max, V}];
+module_opt([<<"ack_freq">>, <<"mod_stream_management">>|_], V) ->
+    [{ack_freq, V}];
+module_opt([<<"resume_timeout">>, <<"mod_stream_management">>|_], V) ->
+    [{resume_timeout, V}];
+module_opt([<<"stale_h">>, <<"mod_stream_management">>|_] = Path, V) ->
+    Stale = parse_section(Path, V),
+    [{stale_h, Stale}];
 module_opt([<<"host">>, <<"mod_vcard">>|_], V) ->
     [{host, b2l(V)}];
-module_opt([<<"ldap_base">>, <<"mod_vcard">>|_], V) ->
+module_opt([<<"search">>, <<"mod_vcard">>|_], V) ->
+    [{search, V}];
+module_opt([<<"matches">>, <<"mod_vcard">>|_], <<"infinity">>) ->
+    [{matches, infinity}];
+module_opt([<<"matches">>, <<"mod_vcard">>|_], V) ->
+    [{matches, V}];
+module_opt([<<"allow_return_all">>, <<"mod_vcard">>|_], V) ->
+    [{allow_return_all, V}];
+module_opt([<<"search_all_hosts">>, <<"mod_vcard">>|_], V) ->
+    [{search_all_hosts, V}];
+module_opt([<<"ldap_uids">>, <<"mod_vcard">>|_] = Path, V) ->
+    Uids = parse_list(Path, V),
+    [{ldap_uids, Uids}];
+module_opt([<<"ldap_vcard_map">>, <<"mod_vcard">>|_] = Path, V) ->
+    Maps = parse_list(Path, V),
+    [{ldap_vcard_map, Maps}];
+module_opt([<<"ldap_search_fields">>, <<"mod_vcard">>|_] = Path, V) ->
+    Fields = parse_list(Path, V),
+    [{ldap_search_fields, Fields}];
+module_opt([<<"ldap_search_reported">>, <<"mod_vcard">>|_] = Path, V) ->
+    Reported = parse_list(Path, V),
+    [{ldap_search_reported, Reported}];
+module_opt([<<"ldap_search_operator">>, <<"mod_vcard">>|_], V) ->
+    [{ldap_search_operator, b2a(V)}];
+module_opt([<<"ldap_binary_search_fields">>, <<"mod_vcard">>|_], V) ->
+    [{ldap_binary_search_fields, V}];
+module_opt([<<"os_info">>, <<"mod_version">>|_], V) ->
+    [{os_info, V}];
+% General options
+module_opt([<<"iqdisc">>|_], V) ->
+    [{iqdisc, b2a(V)}];
+module_opt([<<"backend">>|_], V) ->
+    [{backend, b2a(V)}];
+%% LDAP-specific options
+module_opt([<<"ldap_pool_tag">>|_], V) ->
+    [{ldap_pool_tag, b2a(V)}];
+module_opt([<<"ldap_base">>|_], V) ->
     [{ldap_base, b2l(V)}];
-module_opt([<<"ldap_filter">>, <<"mod_vcard">>|_], V) ->
+module_opt([<<"ldap_filter">>|_], V) ->
     [{ldap_filter, b2l(V)}];
-module_opt([<<"backend">>|_], V) ->
-    [{backend, b2a(V)}].
+module_opt([<<"ldap_deref">>|_], V) ->
+    [{ldap_deref, b2a(V)}];
+%% Riak-specific options
+module_opt([<<"defaults_bucket_type">>|_], V) ->
+    [{defaults_bucket_type, V}];
+module_opt([<<"names_bucket_type">>|_], V) ->
+    [{names_bucket_type, V}];
+module_opt([<<"version_bucket_type">>|_], V) ->
+    [{version_bucket_type, V}];
+module_opt([<<"bucket_type">>|_], V) ->
+    [{bucket_type, V}];
+module_opt([<<"search_index">>|_], V) ->
+    [{search_index, V}].
```

Comment:
Are they used outside `mod_privacy_riak`? Is it possible to limit the path somehow?

Reply:
They are used quite often when `riak` can be chosen as a backend module. I can restrict it by requiring `riak` as a step, i.e. 
```
[modules.mod_offline]
  access_max_user_messages = "max_user_offline_messages"
  backend = "riak"
  bucket_type = "offline"
```
would become 
```
[modules.mod_offline]
  access_max_user_messages = "max_user_offline_messages"
  backend = "riak"
  riak.bucket_type = "offline"
```